src/EDMF_Environment.jl

```diff
@@ -356,11 +353,16 @@ function sgs_quadrature(self::EnvironmentThermodynamics, EnvVar::EnvironmentVari
             # update cloudy/dry variables for buoyancy in TKE
             EnvVar.cloud_fraction.values[k] = outer_env[i_cf]
             self.qt_dry[k] = outer_env[i_qt_dry]
-            self.th_dry[k] = theta_c(self.Ref.p0_half[k], outer_env[i_T_dry])
+            phase_part = TD.PhasePartition(self.qt_dry[k], 0.0, 0.0)
+            self.th_dry[k] =
+                TD.dry_pottemp_given_pressure(param_set, outer_env[i_T_dry], self.Ref.p0_half[k], phase_part)
+
             self.t_cloudy[k] = outer_env[i_T_cld]
             self.qv_cloudy[k] = outer_env[i_qt_cld] - outer_env[i_ql]
             self.qt_cloudy[k] = outer_env[i_qt_cld]
-            self.th_cloudy[k] = theta_c(self.Ref.p0_half[k], outer_env[i_T_cld])
+            phase_part = TD.PhasePartition(self.qt_cloudy[k], 0.0, 0.0)
+            self.th_cloudy[k] =
+                TD.dry_pottemp_given_pressure(param_set, outer_env[i_T_cld], self.Ref.p0_half[k], phase_part)
```

Comment:
```suggestion
            ts = TD.PhaseEquil_pTq(param_set, self.Ref.p0_half[k], outer_env[i_T_cld], self.qt_cloudy[k])
            self.th_cloudy[k] = TD.dry_pottemp(ts)
```
You'll want to test this out, but I believe that this should give the same result (`T, p` and `q_tot` are specified).